core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/WorkflowWorker.scala

```diff
@@ -82,6 +86,9 @@ class WorkflowWorker(
   val dpThread =
     new DPThread(workerConfig.workerId, dp, logManager, inputQueue)
 
+  val inputRecordings =
```

Comment:
change it to `recordedInputs`? also consider moving this into a manager/gateway? it may not belong to a worker?

Reply:
it has to stay in the main thread and the message must be recorded before sending back the ack. I cannot move it to another place.